src/components/PopupWithForm.js

```diff
@@ -13,6 +14,13 @@ export default class PopupWithForm extends Popup {
     });
     return data;
   }
+  renderLoad(isLoading) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = "Save";
+    }
+  }
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => {
```

Comment:
Please, delete `this._popupForm.reset();` from line 29

You need to **reset** the form only  in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more